signer/src/transaction_signer.rs

```diff
@@ -517,10 +517,19 @@ where
         &mut self,
         msg: &message::WstsMessage,
         bitcoin_chain_tip: &model::BitcoinBlockHash,
+        msg_public_key: PublicKey,
     ) -> Result<(), Error> {
         tracing::info!("handling message");
+        let chain_tip_report = self
+            .inspect_msg_chain_tip(msg_public_key, bitcoin_chain_tip)
+            .await?;
```

Comment:
Why not just pass in the report from `handle_signer_message`?

Reply:
Done